src/Components/shareList.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { fb } from '../lib/firebase';
+import { setLocalToken } from '../lib/token.js';
+
+class ShareList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { token: '' };
+
+    this.updateInput = this.updateInput.bind(this);
+  }
+
+  updateInput(event) {
+    this.setState({ token: event.target.value });
+  }
+
+  getExistingList() {
+    const db = fb.firestore();
+    const promise = db.collection(this.state.token).get();
+
+    promise.then(querySnapshot => {
+      //check if value has list
+      if (!querySnapshot.empty) {
+        //if yes, set local token and route to list view
+        setLocalToken(this.state.token);
+        window.location.href = '/list-view';
```

Comment:
I'd suggest using React Router's `history` API. I already mentioned this in my comment in `shareList` component.

- https://reacttraining.com/react-router/web/api/withRouter
- https://reacttraining.com/react-router/web/api/Hooks/usehistory

Reply:
Not a deal breaker though, if you're short on time, leave this as is!

Reply:
One note about `useHistory` - it can't be used in this class based component. You'd have to wrap this component with `withRouter` (a Higher order Component, or HoC)